kv/memberlist/memberlist_client.go

```diff
@@ -585,7 +585,7 @@ func (m *KV) joinMembersOnStartup(ctx context.Context) bool {
 		nodes := m.discoverMembers(ctx, m.cfg.JoinMembers)
 
 		reached, err := m.memberlist.Join(nodes) // err is only returned if reached==0.
-		if err == nil {
+		if err == nil && !(len(m.cfg.JoinMembers) > 0 && len(nodes) == 0) { // fail on error or join members is set, but no members actually resolved
```

Comment:
If `discoverMembers` returns 0 members, then we don't need to try `m.memberlist.Join` at all, but can back off and retry. WDYT?

Reply:
made the optimization and a bit more